### PR TITLE
feat: allow emoji strings to be passed through constructors

### PR DESCRIPTION
**Pull request by ImRodry**
**Please describe the changes this PR makes and why it should be merged:**
This PR adds the feature from #7616 to builder constructors

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/discord.js/src/structures/SelectMenuBuilder.js [hunk 1]
```diff
@@ -2,10 +2,19 @@
...
 
 class SelectMenuBuilder extends BuildersSelectMenuComponent {
-  constructor(data) {
-    super(Transformers.toSnakeCase(data));
+  constructor({ options, ...data }) {
+    super(
+      Transformers.toSnakeCase({
+        options: options.map(({ emoji, ...option }) => ({
+          emoji: emoji && typeof emoji === 'string' ? Util.parseEmoji(emoji) : emoji,
+          ...option,
```
Comment (vladfrangu) [inline packages/discord.js/src/structures/SelectMenuBuilder.js:13]: These two lines should go the other way around (as spreading options will replace the emoji you just made)

File: packages/discord.js/src/structures/ButtonBuilder.js [hunk 2]
```diff
@@ -5,8 +5,10 @@ const Transformers = require('../util/Transformers');
 const Util = require('../util/Util');
 
 class ButtonBuilder extends BuildersButtonComponent {
-  constructor(data) {
-    super(Transformers.toSnakeCase(data));
+  constructor({ emoji, ...data }) {
+    super(
+      Transformers.toSnakeCase({ emoji: emoji && typeof emoji === 'string' ? Util.parseEmoji(emoji) : emoji, ...data }),
```
Comment (vladfrangu) [inline packages/discord.js/src/structures/ButtonBuilder.js:10]: Ditto here

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

File: packages/discord.js/src/structures/ButtonBuilder.js [hunk 3]
```diff
@@ -5,8 +5,10 @@ const Transformers = require('../util/Transformers');
 const Util = require('../util/Util');
 
 class ButtonBuilder extends BuildersButtonComponent {
-  constructor(data) {
-    super(Transformers.toSnakeCase(data));
+  constructor({ emoji, ...data }) {
+    super(
+      Transformers.toSnakeCase({ ...data, emoji: emoji && typeof emoji === 'string' ? Util.parseEmoji(emoji) : emoji }),
```
Comment (kyranet) [inline packages/discord.js/src/structures/ButtonBuilder.js:10]: Is the `emoji &&` check intentionally there to exclude empty strings (`''`) from being parsed? Are they valid?

**Review by kyranet — COMMENTED** [review]
Just a question:

File: packages/discord.js/src/structures/ButtonBuilder.js (hunk 3 above)
Comment (ImRodry) [reply packages/discord.js/src/structures/ButtonBuilder.js:10]: I believe this was actually done to check for truthy values as I may have added the typeof check later on and forgot to remove that part. Should I remove it then or is it fine to keep?

Comment (kyranet) [reply packages/discord.js/src/structures/ButtonBuilder.js:10]: If `''` is somehow valid, then this code is automatically valid, otherwise remove.

**Pull request merged**